libqtile/group.py

```diff
@@ -265,6 +265,12 @@ def add(self, win, focus=True, force=False):
         if focus:
             self.focus(win, warp=True, force=force)
 
+    def swap(self, win, win_other):
+        try:
+            self.layout.swap(win, win_other)
+        except AttributeError:
+            logger.warning("This layout does not support swapping tiled windows")
```

Comment:
This will catch unexpected attribute errors that are raised from within the `swap` method, so it might be better explicitly testing for the existence of the method with `hasattr`. That said, for the sake of self-documenting code, you could add the swap method to the base `Layout` class that does nothing expect error, then you don't need to check whether the method is on the current layout:
```py
    def swap(win, win_other) -> None:
        raise NotImplementedError()
```
If it's always used within a command (here it's nested within `Window.set_position()`, then maybe raising the CommandError is more appropriate, and I think it would then get caught by the command machinery which would log a message saying the command couldn't be executed.